dassl/optim/optimizer.py

```diff
@@ -36,50 +37,50 @@ def build_optimizer(model, optim_cfg):
                 optim, AVAI_OPTIMS
             )
         )
-
-    if staged_lr:
-        if not isinstance(model, nn.Module):
-            raise TypeError(
-                "When staged_lr is True, model given to "
-                "build_optimizer() must be an instance of nn.Module"
-            )
-
-        if isinstance(model, nn.DataParallel):
-            model = model.module
-
-        if isinstance(new_layers, str):
-            if new_layers is None:
-                warnings.warn(
-                    "new_layers is empty, therefore, staged_lr is useless"
+    if param_groups is None:
```

Comment:
Looks like `param_group` and `staged_lr` cannot be used at the same time.

Here we need to throw an error or warning that once `param_group` is given, `staged_lr` shouldn't be True.

Reply:
Yes, at this time, the learning rate needs to be attached to the param_groups by the user.